apps/studio/src/features/sign-in/components/SingpassCallback.tsx

```diff
@@ -36,5 +43,55 @@ export const SingpassCallback = (): JSX.Element => {
     },
   )
 
-  return <FullscreenSpinner />
+  const { isNewUser, redirectUrl } = data
+
+  if (!isNewUser) {
+    return <FullscreenSpinner />
+  }
```

Comment:
do we need this since we already handle this conditional check in `onSuccess`?

Reply:
Yes, the onSuccess route change may not happen instantaneously, so this check prevents the user from seeing the contents below if they are confirmed to be an existing user.